### PR TITLE
Use the cost metrics method

### PR DESCRIPTION
**Pull request by molnard**
On top of https://github.com/nopara73/Sake/pull/19

### CODE REVIEW

**Comment by nopara73** [comment]
Denomination being an output type is very confusing.

File: Sake/Mixer.cs [hunk 1]
```diff
@@ -291,31 +291,51 @@ public IEnumerable<ulong> Decompose(IEnumerable<Money> myInputsParam, IEnumerabl
...
+            var smallestScriptType = Math.Min(ScriptType.P2WPKH.EstimateOutputVsize(), ScriptType.Taproot.EstimateOutputVsize());
+            var maxNumberOfOutputsAllowed = Math.Min(availableVsize / smallestScriptType, 8); // The absolute max possible with the smallest script type.
             var tolerance = (long)Math.Max(loss, 0.5 * MinAllowedOutputAmountPlusFee); // Taking the changefee here, might be incorrect however it is just a tolerance.
 
-            foreach (var (sum, count, decomp) in Decomposer.Decompose(
-                target: (long)myInputSum,
-                tolerance: tolerance,
-                maxCount: Math.Min(maxNumberOfOutputsAllowed, 8),
-                stdDenoms: stdDenoms))
+            if (maxNumberOfOutputsAllowed > 1)
```
Comment (nopara73) [inline Sake/Mixer.cs:303]: What's this?

Comment (nopara73) [reply Sake/Mixer.cs:303]: What if it isn't? How can it be that it isn't?

**Comment by nopara73** [comment]
Let's review it on Monday

**Comment by molnard** [comment]
> Denomination being an output type is very confusing.

It can be also a change. If you have idea for a better naming.

File: Sake/Mixer.cs (hunk 1 above)
Comment (molnard) [reply Sake/Mixer.cs:303]: If availableVsize is small. It is unlikely but it is coming from the coordinator credential issuer. Maybe it is abnormal and we should bail out the round. But to keep the scope on this specific code. It makes no sense to start looking for combinations if I can only register one output.

Comment (molnard) [reply Sake/Mixer.cs:303]: BTW the master contains this condition, Lucas added it long ago.

Comment (nopara73) [reply Sake/Mixer.cs:303]: @lontivero if you make modifications to this file in the master, please make sure to keep Sake in sync

**Comment by nopara73** [comment]
> It can be also a change. If you have idea for a better naming.

What about Denomination? 😄

**Pull request merged**